### ISSUE TITLE
bundle command resolves refs from external files to a component name using underscore as separator

### ISSUE THREAD

**Issue by mishabruml** [Redocly/redocly-cli#1285]
In the case where components with the same key are imported (referred) from different files, then the filename is prefixed to the object key (to avoid collision) but the prefix is separated with a `_`. 

**Describe the bug**

When bundling swagger with complex references, referring to objects keyed with the same key from external files using the `../filename.yml#/objectKey` syntax, the bundled swagger creates component keys for the objects using a syntax like `filename_objectKey` which then introduces an invalid (for aws api gateway) underscore character `_`. You will see errors in aws such as  `Unable to create model for 'filename_objectKey': Model name must be alphanumeric: filename_objectKey`

Fairly sure this is the offending line
https://github.com/Redocly/redocly-cli/blob/eb827254637d3d17a30b2285fad7d37fb5a3318b/packages/core/src/bundle.ts#L392

**To Reproduce**

Given a swagger file and a couple of schema files

`swagger.yaml`
```yaml
openapi: 3.0.1
paths:
  /v1/foo:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: "schemas/foo.yaml#/v1"

  /v1/bar:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: "schemas/bar.yaml#/v1"
```

`schemas/foo.yaml`
```yaml
v1:
  type: object
  required:
    - foo
  properties:
    foo:
      type: string
```

`schemas/bar.yaml`
```yaml
v1:
  type: object
  required:
    - bar
  properties:
    bar:
      type: string
```

Running `npx redocly bundle swagger.yaml -o swagger-bundled.yaml` yeilds

`swagger-bundled.yaml`
```yaml
openapi: 3.0.1
paths:
  /v1/foo:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/v1'
  /v1/bar:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/bar_v1'
components:
  schemas:
    v1:
      type: object
      required:
        - foo
      properties:
        foo:
          type: string
    bar_v1:
      type: object
      required:
        - bar
      properties:
        bar:
          type: string
```

Note the component at `components.schemas.bar_v1`

Suggestion: allow customisation/inhibition of the separator character, to avoid fouling aws alphanumeric rule or other downstream validations. Similar to the customisation of the separation character in the `split` command.

**Comment by tatomyr** [comment]
Thanks for reporting! I'm not certain how to name such an option though.
@lornajane any suggestions?

**Comment by lornajane** [comment]
Maybe it makes sense to also name it `--separator` here? It would need different help text though in the context of this command. Something like "Object key separator used when bundling named resources from different files"

**Comment by mishabruml** [comment]
I think `separator` is a good fit, the character that separates array/string joins is referred to as `separator` by the js `Array.join()` api which is a similar concept https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join#parameters

**Comment by lornajane** [comment]
Where is it used? Could we use `--key-name-separator` or does that make it more confusing?

**Comment by jbsilva** [comment]
I would use it with `--prefix-components-with-info-prop version`.

**Comment by jeremyfiel** [comment]
it may be better to ask AWS to relax this constraint. Adding a customization for an adjacent service could be a slippery slope.

**Comment by RomanHotsiy** [comment]
I agree with @jeremyfiel but still I don't mind adding some generic option. The name for the option is a tricky part though.

I think that the `--key-name-separator` is not conveying the meaning and I would think it's doing something else.
`--prefix-components-separator` is better.

My suggestion would be `--bundle-component-name-separator` but it's too lengthy maybe 😅. Any better ideas?

**Comment by lornajane** [comment]
Related (but not actually fixing the problem): a good description of the rules that AWS gateway expects: https://github.com/stoplightio/spectral/issues/475

**Comment by dasaplan** [comment]
Hi, I would like to add an observation into the discussion, which I think is related but not bound to AWS.

We observe that the redocly cli will generate "schemas-Foo" as a schema identifier, if Foo is defined in multiple files and are not equal. So, "dash" as a separator within the identifier.

In this discussion the observation seems to me is, that it will generate "schemas_Foo". Hence, "underscore" as a seperator.

My pre-assesment from the code is, that spereation is case specific as designed. But I will still need to verify that assumption.

However, as this discussion makes the case that a config for separators would be helpful, I would like to add, that it would be also helpful, if the seperstor would be consistent for schema identifier in general.

Consistency and configuration would also aid use cases like code generation by reducing the complexity of the input.

**Comment by H1Gdev** [comment]
I faced the same issue too...

When referencing components in other files, how about including the file path in the component name ?
(Excluding separators.)

**Comment by tatomyr** [comment]
@H1Gdev why do you think it would be useful to include the file paths?

**Comment by H1Gdev** [comment]
@tatomyr 
In many cases, file paths are used for uniqueness.

**Comment by kanoru3101** [comment]
When bundling, name conflicts are now fixed by adding a number (like `-2`) instead of the file name with an underscore. 

If you want to choose the schema names yourself (for example, to pass AWS API Gateway's alphanumeric rule), use `--component-names-strategy=title` (See [docs](https://redocly.com/docs/cli/commands/bundle#configure-the-component-names-strategy)). It takes names from each schema's title field, and you can control component names.
Closing as resolved.
